### PR TITLE
feat(dismiss-button): adding small variant

### PR DESCRIPTION
**Pull request by jordanstith15**

### CODE REVIEW

**Comment by 321gillian** [comment]
LGTM apart from the token stuff

File: packages/genesys-spark-components/src/components/stable/gux-dismiss-button/gux-dismiss-button.scss [hunk 1]
```diff
@@ -1,7 +1,28 @@
+:host {
+  // all these to be removed when token issues are fixed
```
Comment (jason-evans-genesys) [inline packages/genesys-spark-components/src/components/stable/gux-dismiss-button/gux-dismiss-button.scss:2]: Should we create a JIRA ticket to track this and add a TODO?

File: packages/genesys-spark-components/src/components/stable/gux-dismiss-button/gux-dismiss-button.tsx [hunk 2]
```diff
@@ -30,7 +36,10 @@ export class GuxDismissButton {
     return (
       <gux-button-slot
         accent="ghost"
-        class={this.position == 'inherit' ? 'gux-inherit' : undefined}
+        class={{
+          'gux-inherit': this.position == 'inherit',
```
Comment (jason-evans-genesys) [inline packages/genesys-spark-components/src/components/stable/gux-dismiss-button/gux-dismiss-button.tsx:40]: should we have a `===` instead?

File: packages/genesys-spark-components/src/components/stable/gux-dismiss-button/gux-dismiss-button.scss (hunk 1 above)
Comment (jordanstith15) [reply packages/genesys-spark-components/src/components/stable/gux-dismiss-button/gux-dismiss-button.scss:2]: Waiting for Daragh on that. He said he'd have one likely by the end of the week if he can't get his fixes to work.

**Review by gavin-everett-genesys — APPROVED** [review]

File: packages/genesys-spark-components/src/components/stable/gux-dismiss-button/gux-dismiss-button.tsx [hunk 3]
```diff
@@ -21,6 +24,9 @@ export class GuxDismissButton {
   @Prop()
   position: GuxDismissButtonPosition = 'absolute';
 
+  @Prop()
+  size: GuxDismissButtonSize = 'medium';
```
Comment (jason-evans-genesys) [inline packages/genesys-spark-components/src/components/stable/gux-dismiss-button/gux-dismiss-button.tsx:28]: Just curious if we will only have ever 2 values for this size: `medium` or `small`, or we will have `large` as well in the future?

**Review by daragh-king-genesys — APPROVED** [review]

File: packages/genesys-spark-components/src/components/stable/gux-dismiss-button/gux-dismiss-button.tsx (hunk 2 above)
Comment (jordanstith15) [reply packages/genesys-spark-components/src/components/stable/gux-dismiss-button/gux-dismiss-button.tsx:40]: I'd prefer to change as little code as possible so I'll keep it that way, but I do think in this case it's unlikely to be much of a difference, especially with the variable having specific strings for types.

**Pull request merged**